samples/extensions/dynamic_multisample_rasterization/dynamic_multisample_rasterization.h

```diff
@@ -0,0 +1,131 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "api_vulkan_sample.h"
+
+class DynamicMultisampleRasterization : public ApiVulkanSample
+{
+  public:
+	DynamicMultisampleRasterization();
+	virtual ~DynamicMultisampleRasterization();
+
+  private:
+	std::unique_ptr<vkb::core::Buffer> vertex_buffer = nullptr;
+	std::unique_ptr<vkb::core::Buffer> index_buffer  = nullptr;
+
+	std::unique_ptr<vkb::sg::Scene>    scene;
+	std::vector<VkDescriptorImageInfo> image_infos;
+	std::map<std::string, int32_t>     name_to_texture_id;
+
+	struct SceneNode
+	{
+		vkb::sg::Node    *node;
+		vkb::sg::SubMesh *sub_mesh;
+	};
+	std::vector<SceneNode> scene_nodes_opaque;
+	std::vector<SceneNode> scene_nodes_opaque_flipped;
+	std::vector<SceneNode> scene_nodes_transparent;
+	std::vector<SceneNode> scene_nodes_transparent_flipped;
+
+	struct UniformData
+	{
+		glm::mat4 projection;
+		glm::mat4 view;
+	} uniform_data;
+	std::unique_ptr<vkb::core::Buffer> uniform_buffer;
+
+	VkPipeline            pipeline;
+	VkPipeline            pipeline_inversed_rasterizer;
+	VkPipelineLayout      pipeline_layout;
+	VkDescriptorSet       descriptor_set;
+	VkDescriptorSetLayout descriptor_set_layout;
+
+	// GUI
+	VkPipeline            pipeline_gui;
+	VkPipelineLayout      pipeline_layout_gui;
+	VkDescriptorSet       descriptor_set_gui;
+	VkDescriptorSetLayout descriptor_set_layout_gui;
+	VkDescriptorPool      descriptor_pool_gui;
+
+	ImageData    color_attachment;
+	VkClearValue clear_values[3];
+
+	/**
+	 * @brief List of MSAA levels supported by the platform
+	 */
+	std::vector<VkSampleCountFlagBits> supported_sample_count_list{};
+
+	/**
+	 * @brief Enables MSAA if set to more than 1 sample per pixel
+	 *        (e.g. sample count 4 enables 4X MSAA)
+	 */
+	VkSampleCountFlagBits sample_count{VK_SAMPLE_COUNT_1_BIT};
+
+	bool sample_count_prepared = false;
+
+	struct
+	{
+		glm::mat4 model_matrix;
+
+		glm::vec4 base_color_factor;
+		float     metallic_factor;
+		float     roughness_factor;
+
+		int32_t base_texture_index;
+		int32_t normal_texture_index;
+		int32_t pbr_texture_index;
+
+	} push_const_block;
+
+	struct GUI_settings
+	{
+		int                      sample_count_index = 0;
+		std::vector<std::string> sample_counts;
+	} gui_settings;
+
+  public:
+	virtual void build_command_buffers() override;
+	virtual void request_gpu_features(vkb::PhysicalDevice &gpu) override;
+	virtual bool prepare(const vkb::ApplicationOptions &options) override;
+	// virtual void setup_render_pass() override;
+	virtual void render(float delta_time) override;
+	// virtual void setup_framebuffer() override;
```

Comment:
out-commented functions should be removed.

Reply:
Removed.